From: Steven Chamberlain <steven@pyro.eu.org>
Subject: GL usage for GNU/kFreeBSD
Date: Sat, 18 Oct 2014 13:12:53 +0100

Since mesa 10.3, glx.h fails to compile if GL_GLEXT_LEGACY
is used without also GLX_GLXEXT_LEGACY.

And GNU/kFreeBSD should better use the GNU/Linux VG_API_LINUX
rather than VG_API_FREEBSD anyway.

Index: simgear-2017.1.1/simgear/canvas/ShivaVG/src/shConfig.h
===================================================================
--- simgear-2017.1.1.orig/simgear/canvas/ShivaVG/src/shConfig.h	2017-02-24 19:13:48.635980362 +0100
+++ simgear-2017.1.1/simgear/canvas/ShivaVG/src/shConfig.h	2017-02-24 19:13:48.631980416 +0100
@@ -16,7 +16,7 @@
         #define NOMINMAX
     #endif
 
-#elif defined(linux) || defined(__linux)
+#elif defined(linux) || defined(__linux) || defined(__GLIBC__)
 
     // Linux
     #define VG_API_LINUX
@@ -26,7 +26,7 @@
     // MacOS
     #define VG_API_MACOSX
 
-#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
+#elif defined(__FreeBSD__)
 
     // FreeBSD
     #define VG_API_FREEBSD
Index: simgear-2017.1.1/simgear/canvas/ShivaVG/src/shDefs.h
===================================================================
--- simgear-2017.1.1.orig/simgear/canvas/ShivaVG/src/shDefs.h	2017-02-24 19:13:48.635980362 +0100
+++ simgear-2017.1.1/simgear/canvas/ShivaVG/src/shDefs.h	2017-02-24 19:13:48.631980416 +0100
@@ -171,6 +171,7 @@
 #else
     #define GL_GLEXT_LEGACY /* don't include glext.h */
     #include <GL/gl.h>
+    #define GLX_GLXEXT_LEGACY /* don't include glxext.h */
     #include <GL/glx.h>
 #endif
 
